src/data/matrix/basis.lean

```diff
@@ -124,10 +124,15 @@ variables (i j : n) (c : α) (i' j' : n)
 @[simp] lemma diag_zero (h : j ≠ i) : diag n α α (std_basis_matrix i j c) = 0 :=
 funext $ λ k, if_neg $ λ ⟨e₁, e₂⟩, h (e₂.trans e₁.symm)
 
+@[simp] lemma diag_same : diag n α α (std_basis_matrix i i c) = pi.single i c :=
+by { ext j, by_cases hij : i = j; try {rw hij}; simp [hij] }
+
 variable [fintype n]
 
 lemma trace_zero (h : j ≠ i) : trace n α α (std_basis_matrix i j c) = 0 := by simp [h]
 
+lemma trace_same : trace n α α (std_basis_matrix i i c) = c := by simp
```

Comment:
Since the proof is `simp`, do you really need it? What happens if you remove it?

Reply:
At any rate this lemma should include `std_basis_matrix` in its name

Reply:
You're right, I don't really need it.